app/assets/javascripts/discourse/app/components/sidebar/more-section-links.gjs

```diff
@@ -69,28 +71,75 @@ export default class SidebarMoreSectionLinks extends Component {
     }
   }
 
-  @action
-  registerClickListener() {
-    this.#addClickEventListener();
+  @bind
+  closeOnEscape(event) {
+    if (event.key === "Escape" && this.open) {
+      this.open = false;
+    }
   }
 
-  @action
-  unregisterClickListener() {
-    this.#removeClickEventListener();
+  @bind
+  closeOnFocusOut(event) {
+    this.#lastFocusEvent = event;
+    scheduleOnce("afterRender", this, this.#handleFocusOut);
+  }
+
+  #handleFocusOut() {
+    const container = document.querySelector(
+      ".sidebar-more-section-links-details-content"
+    );
+    const toggleButton = document.querySelector(
+      ".sidebar-more-section-links-details-summary"
+    );
+
+    if (
+      container &&
+      !container.contains(this.#lastFocusEvent.relatedTarget) &&
+      toggleButton !== this.#lastFocusEvent.relatedTarget // focusing out to the toggle shouldn't close the menu
+    ) {
+      this.open = false;
+    }
   }
 
   @action
   toggleSectionLinks(event) {
     event.stopPropagation();
     this.open = !this.open;
+
+    if (this.open) {
+      scheduleOnce("afterRender", this, this.#focusFirstLink);
+    }
   }
 
-  #removeClickEventListener() {
-    document.removeEventListener("click", this.closeDetails);
+  #focusFirstLink() {
+    const firstLink = document.querySelector(
+      ".sidebar-more-section-links-details-content-wrapper a"
+    );
+    if (firstLink) {
+      firstLink.focus();
+    }
   }
 
-  #addClickEventListener() {
+  #addEventListeners() {
     document.addEventListener("click", this.closeDetails);
+    document.addEventListener("keydown", this.closeOnEscape);
+    document.addEventListener("focusout", this.closeOnFocusOut);
+  }
+
+  #removeEventListeners() {
+    document.removeEventListener("click", this.closeDetails);
+    document.removeEventListener("keydown", this.closeOnEscape);
+    document.removeEventListener("focusout", this.closeOnFocusOut);
+  }
+
+  @action
+  registerEventListeners() {
+    this.#addEventListeners();
+  }
+
+  @action
+  unregisterEventListeners() {
+    this.#removeEventListeners();
```

Comment:
I'd remove those and made `addEventListeners`/`removeEventListeners` into non-private actions